Functions/xASL_num2str.m

```diff
@@ -93,24 +99,28 @@
 		DataOut = num2str(DataIn, f);
 	end
 	
-	% If the bConcatenate option is ON, then check if concatenation is needed
-	if bConcatenate
-		if size(DataOut,1) > 1
-			DataOut(1:end-1,(end+1):(end+length(strDelimiter))) = repmat(strDelimiter,[(size(DataOut,1)-1) 1]);
-			DataOut = DataOut';
-			DataOut = DataOut(:)';
-		end
-		% Replace the extra spaces with the delimiter
-		[indStart,indEnd] = regexp(DataOut,'\d{1}\ +');
-		for iSpace = 1:length(indStart)
-			DataOut = [DataOut(1:indStart(iSpace)) strDelimiter DataOut(indEnd(iSpace)+1:end)];
-			indShift = length(strDelimiter) - (indEnd(iSpace)-indStart(iSpace));
-			indStart = indStart + indShift;
-			indEnd   = indEnd + indShift;
-		end
-	end
+    % If the bConcatenate option is ON, then check if concatenation is needed
+    if bConcatenate
+        if size(DataOut,1) > 1
```

Comment:
This seems unnecessary complicated, line 120 below should suffice?

Not sure what you fixed here, it seems the concatenation option was there already?

Reply:
Or did you only insert a tab here? :)

Reply:
Matlab showed the wrong indentation thing so I did a tab 😆

Reply:
As mentioned in the comments I solved it a bit diffently now. Everything becomes a column vector now and later on the rows are merged in the concat stage. If you don't do this, the default concat for a row vector wouldn't add the default separator (minor bug from before).

Reply:
It still kind of always adds a space after arrays, we could get rid of that, too.

And it would also be nice if someone would check the effect on the pipeline, because we use it quite often, right?

Reply:
Yes, we should remove the extra space 'manually'.

Theoretically - I've checked the code - all should be fine. But we will only see upon re-runing 10TestDataSet and import ;)

Reply:
Ah OK cool, I didn't think of the separator

Reply:
Fixed in 8143da2